doc/manual/src/architecture/store/store.md

```diff
@@ -0,0 +1,212 @@
+# Store
+
+A Nix store is a collection of *store objects* with references between them.
```

Comment:
```suggestion
A Nix store is a collection of file system objects with references between them.
```

"Store object" is new to me and collides with `class Store` in the implementation.

I understand the desire to have an abstraction for a store path with references, but as far as I know, this abstraction isn't used anywhere else.

Reply:
"Store object" is terminology from the PhD thesis. It's not commonly used, but this is because no one else ever documented it. They are decidedly different from file system objects, because FSOs don't model references.

> I understand the desire to have an abstraction for a store path with references, but as far as I know, this abstraction isn't used anywhere else.

This was a point of contention, but we deliberated to focus on the principles and separating data model from implementation.